src/Raven.Client/Documents/Session/DocumentQuery.cs

```diff
@@ -908,6 +908,16 @@ private IGraphQuery<T> WithInternal<TOther>(string alias, DocumentQuery<TOther>
             }
 
             WithTokens.AddLast(new WithToken(alias, docQuery.ToString()));
+
+            if (docQuery.TheWaitForNonStaleResults)
```

Comment:
We need to add it to AsyncDocumentQuery also IIRC.
Can you create test for both async and sync?

Reply:
Probably it would be cleaner to do:
```csharp
            if (docQuery.TheWaitForNonStaleResults)
                WaitForNonStaleResults(docQuery.Timeout);
```

Reply:
@ppekrol TheWaitForNonStaleResults and Timeout are both protected not sure if we should expose them, will it be okay to check it with reflection?

Reply:
No, I don't want that

Reply:
No need that, just let's do the async version and add some simple tests

Reply:
how to test it?

Reply:
I've amended your commit. Check what I've did.